daq/topology.py

```diff
@@ -535,9 +552,14 @@ def _generate_main_acls(self):
             self._make_acl_rule(ports=[self.VXLAN_SEC_TRUNK_PORT])]
         acls[self.INCOMING_ACL_FORMAT % self._COUPLER_ACL] = [
             self._make_acl_rule(ports=[self.VXLAN_SEC_TRUNK_PORT], allow=True)]
+        self._add_dot1x_vxlan_acl(acls)
 
         for port_set in range(1, self.sec_port):
             vlan = self._port_set_vlan(port_set)
+            if port_set in self._set_devices:
+                portset_device = next(iter(self._set_devices[port_set]))
+                if portset_device.vlan:
+                    vlan = portset_device.vlan
```

Comment:
try to avoid mutable variables with any significant scope. You can/should rewrite this such that you calculate an intermediate variable and then assign to vlan once (vlan = portset_device_vlan or self._port_set_vlan), or better yet can this logic be folded into the function (_port_set_vlan)?

Reply:
Moved it to _port_set_vlan instead.